tests/test_forms.py

```diff
@@ -1805,6 +1809,101 @@ async def validate_my_slot(
     ]
 
 
+@pytest.mark.parametrize(
+    "my_required_slots, extracted_values, asserted_events",
+    [
+        (
+            ["state", "city"],
+            {"state": "california", "city": "san francisco"},
+            [["state", "CA"], ["city", "San Francisco"], [REQUESTED_SLOT, None]],
+        ),
+        (
+            ["city", "state"],
```

Comment:
I see that in the first test the order of `my_required_slots` is first "state", then "city". In the 2nd test, the order is reversed, was this intentional? if so, why?

Reply:
Yes, this is intentional. This is to test that, with commit eaf8d86, the extracted value of "city" can subsequently influence the extraction of "state" - that is, knowing the city is san francisco should allow us to infer that the state is california without requiring to see the term in user's message.